src/coreclr/src/jit/importer.cpp

```diff
@@ -20809,6 +20846,39 @@ void Compiler::impDevirtualizeCall(GenTreeCall*            call,
             JITDUMP("Sorry, failed to find unboxed entry point\n");
         }
     }
+    // check wheter we have returned an instantiating stub for generic DIM
+    else if (isInterface)
+    {
+        if (requiresInstMethodTableArg)
+        {
+            assert(((SIZE_T)ownerType & CORINFO_CONTEXTFLAGS_MASK) == CORINFO_CONTEXTFLAGS_CLASS);
+            CORINFO_CLASS_HANDLE exactClassHandle = eeGetClassFromContext(ownerType);
```

Comment:
Ha! Just found that we can't use whatever class is in the context:
```csharp
interface I<in T> // Contravariance!
{
    string DefaultTypeOf() => typeof(T).Name;
}

class C : I<object>
{
}

public static class Program
{
    [MethodImpl(MethodImplOptions.AggressiveOptimization)]
    static int Main()
    {
        var c = new C();
        // `ownerType`  is `I<string>`, and we are going to pass it as type handle arg which is wrong
        // we need to pass `I<object>`!
        var dcs = ((I<string>)c).DefaultTypeOf();
        if (dcs != "Object") return 200;
        return 100;
    }
}
```

So I guess I need to return proper method table from `resolveVirtualMethod` along with devirtualized method desc.